deploy-board/deploy_board/webapp/host_views.py

```diff
@@ -186,6 +188,25 @@ def get(self, request, name, stage, hostname):
             'host_details': host_details,
             'duplicate_stage': duplicate_stage,
         })
+      except:
+        log.error(traceback.format_exc())
+        return render(request, 'hosts/host_details.html', {
```

Comment:
Can we reduce to just 1 render after the exception, rather than having 2 calls to render whether it excepted?

Reply:
We spoke briefly about this, sounds like the exception is in the render. I think we should fix the problem by passing the right info into render() if possible, rather than re-render on exception.

Reply:
For example, in this template, you can check if the key exists. If not, then don't render the line. you can use the Django Template language if statement
```fix:
there is a line in host_details.html which tries to get a value from non-existing key:
{% with env=agent_wrapper.env agent=agent_wrapper.agent hostdetailsvalue=host_details.items|lookup:"Phobos Link" %}```

Reply:
Another option is check in the dictionary being passed to `render()`. If the dictionary does not have the key, you could add the key with an empty value, but I don't think this way is as good.

Reply:
agree